File: Tutorials/WPF/DevExpress.MVVM/VB/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports System.Linq

Namespace WpfApplicationMvvm.ViewModels
	Public Class MainViewModel
		Inherits ViewModelBase

		Protected ReadOnly Property DocumentManagerService() As IDocumentManagerService
			Get
				Return GetService(Of IDocumentManagerService)()
			End Get
		End Property
		<Command>
		Public Sub CreateDocument()
			Dim doc As IDocument = DocumentManagerService.CreateDocument("CustomerListView", Nothing, Me)
			doc.Id = String.Format("DocId_{0}", DocumentManagerService.Documents.Count())
			doc.Title = "Customers"
			doc.Show()
		End Sub
	End Class
End Namespace

File: Tutorials/WPF/DevExpress.MVVM/CS/ViewModels/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using System.Linq;

namespace WpfApplicationMvvm.ViewModels {
    public class MainViewModel :ViewModelBase {
        protected IDocumentManagerService DocumentManagerService {
            get { return GetService<IDocumentManagerService>(); }
        }
        [Command]
        public void CreateDocument() {
            IDocument doc = DocumentManagerService.CreateDocument("CustomerListView", null, this);
            doc.Id = string.Format("DocId_{0}", DocumentManagerService.Documents.Count());
            doc.Title = "Customers";
            doc.Show();
        }
    }
}
